src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result = initialValue;
+    let startIndex = 0;
+
+    if (initialValue === undefined) {
+      result = this[0];
+      startIndex++;
```

Comment:
```suggestion
      startIndex = 1;
```

no need to use incrementation here. There are only two possible values you need - either 0 or 1. So better to assign it directly